JS_README.md

```diff
@@ -0,0 +1,115 @@
+
+Primative data structures will be converted automatically. More complex data structures can be converted to native javascript objects via a call to a special function called `json`
+
+``` js
+let qcname = lisp.intern(":name");
+let myList = lisp.list(qcname, 3);
+console.log(myList); // prints { nativeProxy : true }
+console.log(myList.json()); // prints { name: 3 }, defaulting to the assumption this list is a plist
+```
+
+It's important to note that once you convert to a native js object via a call to `json`, your mutations do not affect the lisp layer.
+
+``` js
+let result = lisp.buffer_string();
+result += "my addition"; // This operation did not edit any lisp object.
+let myList = lisp.list(qcname, 3);
+let jsond = myList.json();
+jsond.name = 4; // This did not edit a lisp object.
+```
+
+We expose the async IO functionality included with deno. Users can fetch data async from their local file system, or  the network. They can use that data to interact with the editor. An example would be:
+
+``` js
+const json = fetch("https://api.github.com/users/denoland")
+.then((response) => { return response.json(); });
+
+const txt = Deno.readTextFile("./test.json");
+
+Promise.all([json, text])
+    .then((data) => {
+        let buffer = lisp.get_buffer_create('hello');
+        const current = lisp.current_buffer();
+        lisp.set_buffer(buffer);
+        lisp.insert(JSON.stringify(data[0]));
+        lisp.insert(data[1]);
+        console.log(lisp.buffer_string());
+        lisp.set_buffer(current);
+    });
+```
+This example assumes you have a json file named `test.json` in your current directory.
+
+Lambdas are converted automatically between the two languages via proxying. You can use  this to set timers, or to add hooks. This example features the lisp.symbols object, which returns a proxy to the quoted key. lisp.setq functions similarly to `(setq ....)`
+
+``` js
+lisp.add_hook(lisp.symbols.text_mode_hook, () => {
+    lisp.setq(lisp.symbols.foo, "3");
+});
+
+// Or set timer with arguments...
+lisp.run_with_timer(lisp.symbols.t, 1, (a, b) => {
+    console.log('hello ' + a);
+    lisp.print(b);
+    return 3;
+}, 3, lisp.make.alist({x: 3}));
+```
+
+The user can also define functions via `defun` that will call back into javascript. Like in normal defun, DOCSTRING and the INTERACTIVE decl are optional. This example features both docstring and interactive
+
+``` js
+lisp.defun("my-function", "This is my cool function", {interactive: true, args: "P\nbbuffer"}, (buff) => console.log("Hello Buffer"));
```

Comment:
this is too ugly. Can we have something like this:

```
lisp.defun.my_function(a) { 
     }.interactive(....)
      .comment("foo-bar")
```
(or similar builder pattern).

Reply:
One JS pattern I like is named argumemnts. What do you think of:

``` js
lisp.defun({
    name: "my-function",
    docstring: "My Docstring",
    interactive: true, 
    func: () => { },
});
```

Reply:
Looks good.

Reply:
Added.